### PR TITLE
Add custom navigation to theme

### PR DESCRIPTION
**Pull request by fernando-mc**
This implements a way to have custom navigation links in the nav bar. 

Let me know if this looks appropriate then I'll rebase and bump the version in package.json.

### CODE REVIEW

File: exampleSite/data/navigation.toml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[[links]]
+title = "About"
+url = "/about/"
```
Comment (curttimson) [inline exampleSite/data/navigation.toml]: Should this file exist in the `/en` folder instead?

With this structure you cannot have different translations of "About" for example depending on which language you're viewing the site in.

File: layouts/partials/nav.html [hunk 2]
```diff
@@ -3,6 +3,9 @@
 <nav id="nav">
     <ul class="links">
         <li class="active"><a href='{{ "/" | relLangURL }}'>{{ .Site.Title }}</a></li>
+        {{ range .Site.Data.navigation.links }}
+            <li><a href="{{ .url | relURL }}" {{ if in .url "http" }}target="_blank"{{ end }}>{{ .title }}</a></li>
```
Comment (curttimson) [inline layouts/partials/nav.html]: I'm not 100% sure about having the `target=blank` for external URLs. Some instances might want this to be the same windows (for example this theme could be used as a documentation site which links to another application under a different domain).

Perhaps the target=blank could be a `newwindow` config instead to allow flexibility?

File: exampleSite/content/about.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
++++
+
+### What are we about?
+Well, we're all about about pages. 
+
+Instead of telling you more about a fake company, here's an example of an about page file in the content directory that is referenced by the navigation.toml file in the data folder. 
+
+If you want to link to a page on your hugo blog just make a page like this in the content folder and use a slug like in this markdown file. 
+
+You could theoretically link to exeternal sites from the navigation file in the data directory too!
```
Comment (curttimson) [inline exampleSite/content/about.md]: Small typo.

exeternal -> external

**Comment by curttimson** [comment]
Thanks @fernando-mc !

I've left a few comments - please let me know what you think 🙂

**Comment by fernando-mc** [comment]
@curttimson Your suggestions make sense - I'm just not entirely sure how to implement the multi-language feature and probably wont get around to it very soon. Feel free to take whatever portion of my PR is useful and apply it to a new one. I didn't really need multi-language functionality for my purpose so I didn't figure out how to implement it.

**Pull request closed without merging**